src/hist/plot.py

```diff
@@ -210,16 +218,229 @@ def plot2d_full(
+        ).argmin(axis=0)
+        argnearest = tuple(dim[nearest] for dim in available)
+        scale[missing] = scale[argnearest]
+    counts = values / scale
+    interval_min = scale * stats.chi2.ppf((1 - coverage) / 2, 2 * counts) / 2.0
+    interval_max = scale * stats.chi2.ppf((1 + coverage) / 2, 2 * (counts + 1)) / 2.0
+    interval = np.array([interval_min, interval_max])
+    interval[interval == np.nan] = 0.0  # chi2.ppf produces nan for counts=0
+    return interval
+
+
+def clopper_pearson_interval(
+    num: np.ndarray, denom: np.ndarray, coverage: "Optional[float]" = None
+) -> np.ndarray:
+    """Compute Clopper-Pearson coverage interval for a binomial distribution
+    Parameters
+    ----------
+        num : numpy.ndarray
+            Numerator, or number of successes, vectorized
+        denom : numpy.ndarray
+            Denominator or number of trials, vectorized
+        coverage : float, optional
+            Central coverage interval, defaults to 68%
+    c.f. http://en.wikipedia.org/wiki/Binomial_proportion_confidence_interval
+    """
+    if coverage is None:
+        coverage = stats.norm.cdf(1) - stats.norm.cdf(-1)
+    # Numerator is subset of denominator
+    if np.any(num > denom):
+        raise ValueError(
+            "Found numerator larger than denominator while calculating binomial uncertainty"
+        )
+    interval_min = stats.beta.ppf((1 - coverage) / 2, num, denom - num + 1)
+    interval_max = stats.beta.ppf((1 + coverage) / 2, num + 1, denom - num)
+    interval = np.array([interval_min, interval_max])
+    interval[:, num == 0.0] = 0.0
+    interval[1, num == denom] = 1.0
+    return interval
+
+
+# TODO: Why is return type "Any"?
+def ratio_uncertainty(
+    num: np.ndarray, denom: np.ndarray, uncert_type: str = "poisson"
+) -> Any:
+    with np.errstate(divide="ignore"):
+        ratio = num / denom
+    if uncert_type == "poisson":
+        ratio_uncert = np.abs(poisson_interval(ratio, num / np.square(denom)) - ratio)
+        # TODO: Why does SciPy not work here?
+        # coverage = stats.norm.cdf(1) - stats.norm.cdf(-1)
+        # alpha = 1 - coverage
+        # ratio_uncert = stats.poisson.interval(alpha, ratio)
+    elif uncert_type == "poisson-ratio":
+        # poisson ratio n/m is equivalent to binomial n/(n+m)
+        ratio_uncert = np.abs(clopper_pearson_interval(num, num + denom) - ratio)
+    return ratio_uncert
+
+
+def _fit_callable_to_hist(
+    model: Callable[[np.ndarray], np.ndarray],
+    _hist: hist.BaseHist,
+    likelihood: bool = False,
+) -> "Tuple[np.ndarray, np.ndarray, np.ndarray, Tuple[Tuple[float, ...], np.ndarray]]":
+    """
+    Fit a model, a callable function, to the histogram values.
+    """
+    variances = _hist.variances()
+    if variances is None:
+        raise RuntimeError(
+            "Cannot compute from a variance-less histogram, try a Weight storage"
+        )
+    hist_uncert = np.sqrt(variances)
```

Comment:
```suggestion
    hist_uncert: np.ndarray = np.sqrt(variances)
```

Sadly needed, due to https://github.com/numpy/numpy/blob/c1ce397565398dcf22eda23f37d3c77ffe8f9b13/numpy/__init__.pyi#L2925 

You'll also need to predeclare `model_uncert: np.ndarray`. You can use cast or # type: ignore to "fix" the fact you are setting a typed value with Any.

Reply:
Sorry, can you explain why this is needed in more detail? I believe you, but I don't understand given that things are passing `mypy` and I'm trying to better get the rules for typing.

Reply:
Well, if MyPy is happy, go with it. :) But in general:

```python
from typing import Any

def f(x: Any) -> int:
    return x
```
fails with:
```
tmp.py:4: error: Returning Any from function declared to return "int"
```

It's generally an error to return a more specific type from something that's less specific. However, if you manually set it in a single line, it's happy:

```python
from typing import Any

def f(x: Any) -> int:
    y: int = x
    return y
```
(I think the RHS here must be very simple to be treated like a cast.)
Or if you cast:
```python
import typing
from typing import Any

def f(x: Any) -> int:
    return typing.cast(int, x)
```
Not as fond of casting personally, as it gets in the way of runtime code more than the explicit assignment or a type ignore.

Reply:
Thanks! I really appreciate you taking the time to explain. I'll change this when I'm out of meetings, but this helps me understand better!

Reply:
If MyPy is currently happy, I wouldn't worry too much about it. This is the general idea, though.